core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -68,103 +114,263 @@ public SimpleQueryStringBuilder boost(float boost) {
     public SimpleQueryStringBuilder field(String field) {
-        this.fields.put(field, null);
+        if (Strings.isEmpty(field)) {
+            throw new IllegalArgumentException("Supplied field is null or empty.");
+        }
+        this.fieldsAndWeights.put(field, 1.0f);
         return this;
     }
 
-    /**
-     * Add a field to run the query against with a specific boost
-     */
+    /** Add a field to run the query against with a specific boost. */
     public SimpleQueryStringBuilder field(String field, float boost) {
-        this.fields.put(field, boost);
+        if (Strings.isEmpty(field)) {
+            throw new IllegalArgumentException("Supplied field is null or empty.");
+        }
+        this.fieldsAndWeights.put(field, boost);
         return this;
     }
 
-    /**
-     * Specify a name for the query
-     */
-    public SimpleQueryStringBuilder queryName(String name) {
-        this.queryName = name;
+    /** Add several fields to run the query against with a specific boost. */
+    public SimpleQueryStringBuilder fields(Map<String, Float> fields) {
+        this.fieldsAndWeights.putAll(fields);
         return this;
     }
 
-    /**
-     * Specify an analyzer to use for the query
-     */
+    /** Returns the fields including their respective boosts to run the query against. */
+    public Map<String, Float> fields() {
+        return this.fieldsAndWeights;
+    }
+
+    /** Specify an analyzer to use for the query. */
     public SimpleQueryStringBuilder analyzer(String analyzer) {
         this.analyzer = analyzer;
         return this;
     }
 
+    /** Returns the analyzer to use for the query. */
+    public String analyzer() {
+        return this.analyzer;
+    }
+
     /**
      * Specify the default operator for the query. Defaults to "OR" if no
-     * operator is specified
+     * operator is specified.
      */
     public SimpleQueryStringBuilder defaultOperator(Operator defaultOperator) {
-        this.operator = defaultOperator;
+        this.defaultOperator = defaultOperator;
         return this;
     }
 
+    /** Returns the default operator for the query. */
+    public Operator defaultOperator() {
+        return this.defaultOperator;
+    }
+
     /**
-     * Specify the enabled features of the SimpleQueryString.
+     * Specify the enabled features of the SimpleQueryString. Defaults to ALL if
+     * none are specified.
      */
     public SimpleQueryStringBuilder flags(SimpleQueryStringFlag... flags) {
-        int value = 0;
-        if (flags.length == 0) {
-            value = SimpleQueryStringFlag.ALL.value;
-        } else {
+        if (flags != null && flags.length > 0) {
+            int value = 0;
             for (SimpleQueryStringFlag flag : flags) {
                 value |= flag.value;
             }
+            this.flags = value;
+        } else {
+            // set flags to invalid value so we can complain on validation below
+            this.flags = MISSING_FLAGS;
```

Comment:
given what we do in field methods, I wonder if we should do the same here (fail fast) rather than postponing the failure, which requires a MISSING_FLAGS constant etc. What do you prefer?